### DISCUSSION TITLE
Handle multiple triplets simultaneously in manifest mode

### DISCUSSION THREAD

**Discussion by Anton-V-K** [microsoft/vcpkg#27277] [Q&A] [answered]
I want to freeze the version of one dependency (`openssl`), and the approach with `overrides` in manifest mode (as suggested in https://github.com/microsoft/vcpkg/discussions/26723#discussioncomment-3593425 ) seems to be the way to go.
Now I'm wondering is it posible to keep several platforms in one directory when the manifest mode is active?
Without the manifest I could handle it with one command like
```
vcpkg install asio:x64-windows openssl-windows:x64-windows-static ... asio:x86-windows openssl-windows:x86-windows-static ...
```
So it is a mix of triplets.
Unfortunately _In manifest mode, `vcpkg install` does not support individual package arguments_ :( Is this a known limitation? Will it be addressed in future versions?
I don't use CMake for building (just VS2022 projects), so workarounds like suggested in https://github.com/microsoft/vcpkg/discussions/25033 aren't applicable.
I have the manifest `vcpkg.json` like this
```
{
    "name": "plugins",
    "version": "1.0.0",
    "builtin-baseline": "a69b65229b3956b7f45abd81671b7330114bcaad",
    "dependencies": [
        {
            "name": "asio",
            "platform": "windows & (x64 | x86)"
        },
        {
            "name": "openssl-windows",
            "platform": "windows & (x64 | x86) & static"
        }
    ],
    "overrides": [
        {
            "name": "openssl-windows",
            "version-string": "1.1.1h#1"
        }
    ]
}
```
I can't grab all dependencies in one go with `vcpkg install --triplet x64-windows --triplet x64-windows-static --triplet x86-windows
--triplet x86-windows-static` (only one triplet can be specified in the command line - see the related discussion https://github.com/microsoft/vcpkg/discussions/17760), and getting them step-by-step this way
```
vcpkg install --triplet x64-windows
vcpkg install --triplet x64-windows-static
vcpkg install --triplet x86-windows
vcpkg install --triplet x86-windows-static
```
just leaves the last triplet alive.
Probably I'm overlooking something...
Is there an easy way to achieve the goal?
Just to sum up:
1. Put all dependencies into one or two (x64 and x86) subdirs (there is a command line option to separate directories - `--x-install-root=x64-windows-static` though it is marked as experimental)
2. Fix the version of one (or more) dependency (refer to https://github.com/microsoft/vcpkg/discussions/26723#discussioncomment-3593425).
3. Have a mix of static and non-static libs within one platform (so far I'm not sure whether I can use only static or only non-static libs).

**Comment by autoantwort** [comment]
You could use the msbuild integration.

**Reply by Anton-V-K** [reply]
Yeah, both VS2019 and VS2022 have good built-in support of vcpkg, and with manifest they can easily handle 1-st and 2-nd points without extra efforts from my side.

**Comment by Osyotr** [comment, marked as answer]
> 3. Have a mix of static and non-static libs within one platform 

https://github.com/microsoft/vcpkg/blob/master/docs/users/triplets.md#per-port-customization

Installing multiple variants from one manifest is possible, just change installed directory (i.e. `vcpkg install --x-install-root=<path>`), but I'd recommend using some form of build-system integration as suggested above.

**Reply by Anton-V-K** [reply]
The option `--x-install-root=<path>` does the trick for me.
And the integration-based approach isn't an option in my case since the project uses local vcpkg "storage", which may potentially conflict with other projects if integrated user-wide.

**Reply by Osyotr** [reply]
You can integrate vcpkg project-wide.

**Reply by Anton-V-K** [reply]
I'm only aware of user-wide integration with `vcpkg integrate install`.
Anyway I've managed to find a guide like [this one about msbuild integration](https://github.com/microsoft/vcpkg/blob/master/docs/users/buildsystems/msbuild-integration.md#linked-nuget-package) (which supposedly works per-project), but this approach is pre-faced with a piece of advice like
> Note: This approach is not recommended for new projects, since it makes them difficult to share with others. For a portable, self-contained NuGet package, see the [export command](https://github.com/microsoft/vcpkg/blob/master/docs/users/buildsystems/export-command.md)

And the linked instruction doesn't provide enough details about the approach.
Eventually I've ended up with a hand-made .props-file and a batch file for a pre-build step to set up vcpkg.
